core/src/main/java/org/elasticsearch/rest/BytesRestResponse.java

```diff
@@ -116,31 +119,23 @@ public RestStatus status() {
 
     private static final Logger SUPPRESSED_ERROR_LOGGER = ESLoggerFactory.getLogger("rest.suppressed");
 
-    private static XContentBuilder convert(RestChannel channel, RestStatus status, Exception e) throws IOException {
-        XContentBuilder builder = channel.newErrorBuilder().startObject();
-        if (e == null) {
-            builder.field("error", "unknown");
-        } else if (channel.detailedErrorsEnabled()) {
-            final ToXContent.Params params;
-            if (channel.request().paramAsBoolean("error_trace", !ElasticsearchException.REST_EXCEPTION_SKIP_STACK_TRACE_DEFAULT)) {
-                params =  new ToXContent.DelegatingMapParams(
-                        Collections.singletonMap(ElasticsearchException.REST_EXCEPTION_SKIP_STACK_TRACE, "false"), channel.request());
+    private static XContentBuilder build(RestChannel channel, RestStatus status, Exception e) throws IOException {
+        ToXContent.Params params = channel.request();
+        if (params.paramAsBoolean("error_trace", !REST_EXCEPTION_SKIP_STACK_TRACE_DEFAULT)) {
+            params =  new ToXContent.DelegatingMapParams(singletonMap(REST_EXCEPTION_SKIP_STACK_TRACE, "false"), params);
+        } else if (e != null) {
+            Supplier<?> messageSupplier = () -> new ParameterizedMessage("path: {}, params: {}",
+                    channel.request().rawPath(), channel.request().params());
+
+            if (status.getStatus() < 500) {
+                SUPPRESSED_ERROR_LOGGER.debug(messageSupplier, e);
             } else {
-                if (status.getStatus() < 500) {
-                    SUPPRESSED_ERROR_LOGGER.debug(
-                            (Supplier<?>) () -> new ParameterizedMessage("path: {}, params: {}",
-                                    channel.request().rawPath(), channel.request().params()), e);
-                } else {
-                    SUPPRESSED_ERROR_LOGGER.warn(
-                            (Supplier<?>) () -> new ParameterizedMessage("path: {}, params: {}",
-                                    channel.request().rawPath(), channel.request().params()), e);
-                }
-                params = channel.request();
+                SUPPRESSED_ERROR_LOGGER.warn(messageSupplier, e);
             }
-            ElasticsearchException.renderException(builder, params, e);
-        } else {
-            builder.field("error", simpleMessage(e));
         }
+
+        XContentBuilder builder = channel.newErrorBuilder().startObject();
+        ElasticsearchException.generateFailureXContent(builder, params, e, channel.detailedErrorsEnabled());
         builder.field("status", status.getStatus());
         builder.endObject();
```

Comment:
I agree it's unfortunate, specially in the context of `BytesRestResponse` or `MultiSearchResponse`. 

But on the other hand a `status` field is not always associated to the exception rendered using `ElasticsearchException.generateFailureXContent()` like in `MultiSearchTemplateResponse`, so would it make sense to add rendering logic of the status in `generateFailureXContent` and parsing logic in `failureFromXContent`?

Same remark applies for `generateThrowableXContent`, status is sometime rendered as part of the enclosing object (see DefaultShardOperationFailedException or BulkItemResponse) so custom parsing logic will be needed anyway.

But I don't have a string feeling here. I feel like we shouldn't change the current API to make it uniform as it is not clear how the `status` value associated to `ElasticsearchException` is derived (looks like `ExceptionsHelper.status(Throwable)` is used most of the time but it looks like the status value is sometimes excplicitly set to something else).